nodes/src/nodes/embedding_video/IInstance.py

```
@@ -0,0 +1,216 @@
+            self._mime_type = mimeType
+
+        elif action == AVI_ACTION.WRITE:
+            if self._video_data is not None:
+                self._video_data += buffer
+
+        elif action == AVI_ACTION.END:
+            if self._video_data is not None and len(self._video_data) > 0:
+                self._process_video(bytes(self._video_data))
+            self._video_data = None
+
+    def _process_video(self, video_bytes: bytes):
```

Ruff (0.15.7)

[warning] Too many statements (64 > 50)

(PLR0915)